xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-api/src/main/java/org/xwiki/notifications/page/events/PageNotificationEventListener.java

```diff
@@ -0,0 +1,170 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.notifications.page.events;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.notifications.page.PageNotificationEvent;
+import org.xwiki.notifications.page.PageNotificationEventDescriptor;
+import org.xwiki.notifications.page.PageNotificationEventDescriptorContainer;
+import org.xwiki.observation.AbstractEventListener;
+import org.xwiki.observation.ObservationManager;
+import org.xwiki.observation.event.AllEvent;
+import org.xwiki.observation.event.Event;
+import org.xwiki.security.authorization.AuthorExecutor;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+/**
+ * Default implementation of {@link PageNotificationEventDescriptorContainer}.
+ *
+ * @version $Id$
+ * @since 9.4RC1
+ */
+@Component
+@Singleton
+@Named(PageNotificationEventListener.NAME)
+public class PageNotificationEventListener extends AbstractEventListener
+{
+    /**
+     * The listener name.
+     */
+    public static final String NAME = "Page Notification Event Listener";
+
+    @Inject
+    private ObservationManager observationManager;
+
+    @Inject
+    private PageNotificationEventDescriptorContainer pageNotificationEventDescriptorContainer;
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Inject
+    private AuthorExecutor authorExecutor;
+
+    private class VelocityTemplateBooleanEvaluator implements Callable<Boolean>
+    {
+        private String content;
+
+        private Provider<XWikiContext> contextProvider;
+
+        /**
+         * Constructs a {@link VelocityTemplateBooleanEvaluator}.
+         *
+         * @param contextProvider a reference to the XWikiContext provider component
+         * @param content the content of the template that will be evaluated
+         */
+        VelocityTemplateBooleanEvaluator(Provider<XWikiContext> contextProvider, String content)
+        {
+            this.contextProvider = contextProvider;
+            this.content = content;
+        }
+
+        @Override
+        public Boolean call() throws Exception
+        {
+            return contextProvider.get().getWiki().evaluateVelocity(content, "page-notification").contains("true");
```

Comment:
1. In my opinion, it's not the job of the `VelocityTemplateBooleanEvaluator` to also check that the result is `true`. The velocity template evaluator should be more generic and shared with the other use-case.
1. Also check if you can do this using a lambda (so you have less code duplication)
1. You rely to much on contains(). It would be safer to do result.trim().equals("true").